File: src/main/java/model/pieces/Knight.java
package model.pieces;

import model.moves.Move;
import model.util.Colour;
import model.util.Offset;

public class Knight extends OffsetPiece {
    /**
     * The Offsets where the Knight can be moved. In an L shape
     */
    private static final Offset[] OFFSETS = {
            new Offset(-1, -2),
            new Offset(-2, -1),
            new Offset(-2, 1),
            new Offset(-1, 2),
            new Offset(1, 2),
            new Offset(2, 1),
            new Offset(2, -1),
            new Offset(1, -2)
    };

    public Knight(Colour colour) {
        super(colour);
    }

    @Override
    Offset[] getOffsets() {
        return OFFSETS;
    }

    @Override
    int getUnicodeWhite() {
        return 9816;
    }

    @Override
    int getUnicodeBlack() {
        return 9822;
    }

    @Override
    public int getUnsignedValue() {
        return 3;
    }

    @Override
    public void notifyMoveComplete(Move move) {
    }

    @Override
    public void notifyMoveUndo(Move move) {
    }

    @Override
    String getName() {
        return "Knight";
    }
}

File: chess-engine/src/main/java/model/pieces/Knight.kt
package model.pieces

import model.moves.Move
import model.util.Colour
import model.util.Offset

internal class Knight(colour: Colour) : OffsetPiece(colour) {

    override val offsets: Array<Offset>
        get() = OFFSETS

    override val unicodeWhite: Int
        get() = 9816

    override val unicodeBlack: Int
        get() = 9822

    override val unsignedValue: Int
        get() = 3

    override val name: String
        get() = "Knight"

    override fun notifyMoveComplete(move: Move) {}

    override fun notifyMoveUndo(move: Move) {}

    companion object {
        /**
         * The Offsets where the Knight can be moved. In an L shape
         */
        private val OFFSETS = arrayOf(Offset(-1, -2), Offset(-2, -1), Offset(-2, 1), Offset(-1, 2), Offset(1, 2), Offset(2, 1), Offset(2, -1), Offset(1, -2))
    }
}
